### PR TITLE
Example subplots share axes

### LINKED ISSUE #1686: [ENH ]Use sharex=True and sharey=True in some gallery examples
In a large fraction of gallery examples, we use different subplots to display different filtered versions of an image. For such cases, it would be nice to use the keyword arguments of ``plt.subplots`` ``sharex=True, sharey=True`` so that zooming in one of the subplots also updates the other ones.

### PR DESCRIPTION
**Pull request by MartinSavc**
Fixes #1686

### CODE REVIEW

File: doc/examples/applications/plot_coins_segmentation.py [hunk 1]
```diff
@@ -35,7 +35,9 @@
 
 """
 
-fig, (ax1, ax2) = plt.subplots(1, 2, figsize=(6, 3))
+fig = plt.figure(figsize=(6,3))
+ax1 = fig.add_subplot(1, 2, 1, adjustable='box-forced')
+ax2 = fig.add_subplot(1, 2, 2, sharex = ax1, sharey = ax1, adjustable='box-forced')
```
Comment (JDWarner) [inline doc/examples/applications/plot_coins_segmentation.py]: We recently standardized the plot creation API to `fig, (ax0, ax1, ...) = plt.subplots(...)`, which is the preferred Matplotlib style. There are a couple different ways this is implemented in this PR; I prefer the style shown in `plot_morphology.py` to the style here as it continues the use of `plt.subplots`.

Comment (MartinSavc) [inline doc/examples/applications/plot_coins_segmentation.py]: Ok. I'll change these back to 
```python
fig, (ax1, ax2) = plt.subplots(1, 2, figsize=(6, 3), sharex=True, sharey=True, subplot_kw={'adjustable':'box-forced'} )
```
What about the specific cases wher plt.subplots does not seem flexible enough? There are two specific cases:
* sharing axes across figures
* sharing only specific axes in a figure - for example the first column

**Comment by emmanuelle** [comment]
Hi @MartinSavc sorry for taking so long to answer you. In more complex cases where only one axes could be shared, I would suggest not to share axes at all, for the sake of code simplicity in our gallery examples. It's always a difficult trade-off between readability and efficiency/more features... If you don't mind reverting the few examples where you could not use subplots, I think it would be better to improve the readability. If you prefer, you can keep one example corresponding to each case (sharing only one axes, or across figures), in order to document that this is possible.

However, I learnt a few things looking at the different possibilities for sharing axes in your examples, and I think that this could make a nice addition to the user guide. I think we're missing an "Image visualization" section in http://scikit-image.org/docs/stable/user_guide.html, where these explanations on sharing axes could nicely fit.

Thank you very much for your contribution !

**Comment by MartinSavc** [comment]
Ill follow your recommendations - one example for cross figure sharing and sharing specific axes. Ill revert the rest to the requested standard.

**Comment by tacaswell** [comment]
If is better to use `fig.add_subplot` rather than `plt.subplot()`.

**Comment by stefanv** [comment]
Please don't include merge commits in PRs.

**Comment by MartinSavc** [comment]
Is this better?

**Comment by stefanv** [comment]
Much better!  I would still have compressed those "change subplot" commits into one, but I won't be (even more) nitcpicky :)  @emmanuelle I'll leave this one to you to merge when you're ready.

**Comment by emmanuelle** [comment]
Thanks a lot @MartinSavc ! It took some time, but thank you very much for your contribution!

**Comment by stefanv** [comment]
:+1:

**Pull request merged**